File: src/CSharp/290.WordPattern.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace PeefyLeetCode.WordPattern
{
    public class Solution {
    public bool WordPattern(string pattern, string str) {
        str = str.Replace(' ', '|');
        String[] strs = str.Split("|");
        int strscount = strs.Length;
        int pcount = pattern.Length;
        if (strscount != pcount)
            return false;
        var hash = new Dictionary<char, string>();
        for (int i = 0; i < pcount; i++) {
            char c = pattern[i];
            String s = strs[i];
            if (hash.ContainsKey(c) == false){
                if (hash.ContainsValue(s) == true) {
                    return false;
                }
                hash.Add(c, s);
            }
            else{
                if (hash[c] != s)
                    return false;
            }
        }
        return true;
    }
}
}

File: src/Java/290.WordPattern.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.Map;

class Solution {
    public boolean wordPattern(String pattern, String str) {
        str = str.replace(' ', '|');
        String[] strs = str.split("|");
        int strscount = strs.length;
        int pcount = pattern.length();
        if (strscount != pcount)
            return false;
        HashMap<Character, String> hash = new HashMap<>();
        for (int i = 0; i < pcount; i++) {
            Character c = pattern.charAt(i);
            String s = strs[i];
            if (hash.containsKey(c) == false){
                if (hash.containsValue(s) == true) {
                    return false;
                }
                hash.put(c, s);
            }
            else{
                if (hash.get(c) != s)
                    return false;
            }
        }
        return true;
    }
}
